components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicClientAuthenticationHandler.java

```diff
@@ -105,10 +108,26 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
                                     + " with client secret.");
                         }
 
-                        authenticationContext.addProperty(Constants.AUTH_CONTEXT_OAUTH_APP_PROPERTY,
-                                OAuth2Util.getAppInformationByClientId(clientId));
+                        String appTenant = ((AuthenticationContext) messageContext).getAuthenticationRequest()
+                                .getRequest().getParameter("appTenant");
+                        OAuthAppDO oAuthAppDO;
+
+                        /*
+                            When the app tenant is not provided, it tries to retrieve an app for the tenant
+                            in the request path. Even though there could be a scenario for cross tenant access,
+                            the request should be authenticated with a client in the requested tenant. If not,
+                            should pass the app tenant as a query parameter.
+                         */
+                        if (StringUtils.isNotBlank(appTenant)) {
+                            oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId, appTenant);
+                        } else {
+                            oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
+                            appTenant = oAuthAppDO.getAppOwner().getTenantDomain();
```

Comment:
Can the app owner be null? For example, think the user 'A' created an app called 'App1'. So the app owner for the application 'App1' will be user 'A'. If we delete the user 'A' what will happen to the app owner param?

Reply:
No. As per the implementation in [DAO method](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/481052bbb0e6770739084f430228c2119dcaae44/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java#L381-L388) and [SQL query](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/SQLQueries.java#L94), it just retrieves the user info in `IDN_OAUTH_CONSUMER_APPS` table and return the response. Tested the flow and deleting the user doesn't remove the user info in this table. Hence the appTenant will be returned even though the user is deleted.